mne/preprocessing/ica.py

```diff
@@ -548,8 +548,7 @@ def _transform_epochs(self, epochs, concatenate):
         if not hasattr(self, 'mixing_matrix_'):
             raise RuntimeError('No fit available. Please fit ICA')
 
-        picks = pick_types(epochs.info, include=self.ch_names, exclude=[],
-                           ref_meg=False)
+        picks = [epochs.ch_names.index(k) for k in self.ch_names]
```

Comment:
this is what is being used in `_transform_raw` cf. https://github.com/mne-tools/mne-python/pull/2146#discussion-diff-30985610L529. it basically only selects the channels from the ICA object.